### PR TITLE
Support promulgated filter on search

### PR DESCRIPTION
**Pull request by mhilton**

### CODE REVIEW

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore/850/
Test PASSed.

File: internal/charmstore/search.go [hunk 1]
```diff
@@ -499,7 +499,7 @@ func createSearchDSL(sp SearchParams) elasticsearch.QueryDSL {
 			Modifier: "ln2p",
 		},
 		elasticsearch.BoostFactorFunction{
-			Filter:      ownerFilter(""),
```
Comment (urosj) [inline internal/charmstore/search.go]: ownerFilter should still be boosted. We don't want to break the functionality.

File: internal/charmstore/search_test.go [hunk 2]
```diff
@@ -394,6 +394,55 @@ var searchTests = []struct {
...
+			Text: "",
+			Filters: map[string][]string{
+				"promulgated": {"1"},
+			},
+		},
+		results: []*router.ResolvedURL{
+			exportTestCharms["wordpress"],
+			exportTestCharms["mysql"],
+			exportTestBundles["wordpress-simple"],
+		},
```
Comment (urosj) [inline internal/charmstore/search_test.go]: Could we add test that owner= and promulgated=1 return the same results? We must rely on keeping owner= around for now.

**Comment by urosj** [comment]
:+1:

File: internal/charmstore/search_test.go (hunk 2 above)
Comment (mhilton) [inline internal/charmstore/search_test.go]: It is hard to test that they have the same results without doing what's here and explicitly listing them to be the same.

File: internal/charmstore/search.go (hunk 1 above)
Comment (mhilton) [inline internal/charmstore/search.go]: This filter was to boost promulgated entities, using promulgatedFilter just makes it clearer what is being boosted.

File: internal/charmstore/search.go [hunk 3]
```diff
@@ -610,7 +611,7 @@ func nameFilter(value string) elasticsearch.Filter {
 // owner taken from the URL.
 func ownerFilter(value string) elasticsearch.Filter {
 	if value == "" {
-		return elasticsearch.ExistsFilter("PromulgatedURL")
+		return promulgatedFilter("1")
```
Comment (rogpeppe) [inline internal/charmstore/search.go:671]: Are there any other boolean filters?
I wonder if searching for promulgated=yes should really
search for non-promulgated charms?

**Comment by rogpeppe** [comment]
LGTM with one reservation above.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore/865/
Test PASSed.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore/866/
Test PASSed.

**Comment by mhilton** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/charmstore-merge

**Pull request merged**